tests/test_angle.py

```diff
@@ -46,8 +46,9 @@ def test_harmonic_angle(device, dtype):
     ene.backward()
     grads = [coords.grad.clone().detach()]
 
-    for c, g, gref in zip(['coord'], grads, grads_ref):
-        assert torch.allclose(g, gref, atol=1e-5), f'Gradient {c} not the same'
+    grad_atol = 1e-5 if dtype is torch.float64 else 1e-2
+    for c, g, gref in zip(['coord', 'theta0', 'k'], grads, grads_ref):
+        assert torch.allclose(g, gref, atol=grad_atol), f'Gradient {c} not the same'
```

Comment:
Here I specified a tolerance in `torch.allclose` for different precisions. It seems that the single precision needs a really loose criteria.